include/client.h

```diff
@@ -1026,6 +1051,132 @@ class Client
+        *            and use_list_ensemble_prefix() for more details.
+        *   \param old_name The old list name
+        *   \param new_name The new list name
+        *   \throw SmartRedis::Exception if the command fails
+        */
+        void rename_list(const std::string& src_name,
+                         const std::string& dest_name);
+
+        /*!
+        *   \brief Get the length of an aggregation list
+        *   \param list_name The list name
+        *   \throw SmartRedis::Exception if the command fails
+        */
+        int get_list_length(const std::string& list_name);
+
+        /*!
+        *   \brief Poll list length until length is greater or equal
+        *          to the provided length.  If maximum number of
+        *          attemps is exceeded, false is returned.
+        *   \details The aggregation list key used to check for list length
+        *            may be formed by applying a prefix to the supplied
+        *            name. See set_data_source() and use_list_ensemble_prefix()
+        *            for more details.
+        *   \param name The name of the list
+        *   \param list_length The desired length of the list
+        *   \param poll_frequency_ms The time delay between checks,
+        *                            in milliseconds
+        *   \param num_tries The total number of times to check for the name
+        *   \returns Returns true if the list is found with a length greater
+        *            than or equal to the provided length, otherwise false
+        *   \throw SmartRedis::Exception if poll list length command fails
+        */
+        bool poll_list_length(const std::string& name, int list_length,
+                              int poll_frequency_ms, int num_tries);
+
+        /*!
+        *   \brief Get datasets from an aggregation list
+        *   \details The aggregation list key used to retrieve datasets
+        *            may be formed by applying a prefix to the supplied
+        *            name. See set_data_source() and use_list_ensemble_prefix()
+        *            for more details.  An empty or nonexistant
+        *            aggregation list returns an empty vector.
+        *   \param list_name The name of the aggregation list
+        *   \returns A vector containing DataSet objects.
+        *   \throw SmartRedis::Exception if retrieval fails.
+        */
+        std::vector<DataSet> get_datasets_from_list(const std::string& list_name);
+
+        /*!
+        *   \brief Get a range of datasets (by index) from an aggregation list
+        *   \details The aggregation list key used to retrieve datasets
+        *            may be formed by applying a prefix to the supplied
+        *            name. See set_data_source()  and use_list_ensemble_prefix()
+        *            for more details.  An empty or nonexistant aggregation
+        *            list returns an empty vector.  If the provided
```

Comment:
Nonexistent list should throw a runtime exception

Reply:
The group thought that throwing a runtime exception because the list is temporarily empty will make it hard to distinguish between an empty list (which might be useful when dividing work among processes for training) and an execution error.  With future workflows in mind, the group decided to align with the Redis standard of returning an empty vector.